ops/lib.rs

```diff
@@ -136,15 +145,23 @@ fn codegen_v8_async(core: &TokenStream2, f: &syn::ItemFn) -> TokenStream2 {
     };
 
     #core::_ops::queue_async_op(scope, async move {
-      let result = Self::call::<#type_params>(state, #args_tail).await;
+      let result = Self::call::<#type_params>(#args_head #args_tail).await;
       (promise_id, op_id, #core::_ops::to_op_result(get_class, result))
     });
   }
 }
 
 /// Generate the body of a v8 func for a sync op
 fn codegen_v8_sync(core: &TokenStream2, f: &syn::ItemFn) -> TokenStream2 {
-  let (arg_decls, args_tail) = codegen_args(core, f, 1, 1);
+  let arg0 = f.sig.inputs.first();
+  let uses_opstate = arg0.map(is_mut_ref_opstate).unwrap_or_default();
+  let args_head = if uses_opstate {
+    quote! { op_state, }
+  } else {
+    quote! {}
+  };
+  let rust_i0 = if uses_opstate { 1 } else { 0 };
+  let (arg_decls, args_tail) = codegen_args(core, f, rust_i0, 1);
```

Comment:
```suggestion
  let (arg_decls, args_tail) = codegen_args(core, f, usize::from(uses_opstate), 1);
```

Reply:
I think coercing bools to arg offsets could hinder maintainability more than the verbosity of the if expr